src/error.rs

```diff
@@ -693,6 +698,61 @@ impl WriteFailure {
     }
 }
 
+/// An error that occurred during a GridFS operation.
+#[derive(Clone, Debug)]
+#[allow(missing_docs)]
+#[non_exhaustive]
+pub enum GridFsErrorKind {
+    /// The file with the given identifier was not found.
+    #[non_exhaustive]
+    FileNotFound { identifier: GridFsFileIdentifier },
```

Comment:
This is a little convoluted but the `FileNotFound` error can be returned from either `download` (which has an id) or `download_by_name` (which has a filename), and it didn't seem much better to have separate error cases for those.